bluepyopt/ephys/simulators.py

```diff
@@ -182,3 +182,161 @@ def __init__(self, message, original):
+class LFPySimulator(object):
+
+    """LFPy simulator"""
+
+    def __init__(self, LFPyCellModel, electrode=None, cvode_active=True,
+                 cvode_minstep=None, random123_globalindex=None,
+                 mechanisms_directory=None):
+        """Constructor
+        Args:
+            LFPyCellModel (LFPyCellModel): the LFPy cell model
+            electrode (MEAutility.MEA): the MEAutility probe used to compute
+                extracellular signals
+            cvode_active (bool): should neuron use the variable time step
+                integration method
+            cvode_minstep (float): the minimum time step allowed for a cvode
+                step. Default is 0.0.
+            random123_globalindex (int): used to set the global index used by
+                all instances of the Random123 instances of Random
+            mechanisms_directory (str): path to the parent directory of the
+                directory containing the mod files. If the mod files are in
+                "./data/mechanisms", then mechanisms_directory should be
+                "./data/".
+        """
+
+        self.LFPyCellModel = LFPyCellModel
+        self.electrode = electrode
+        self.effective_electrode = electrode
+
+        self.lfpyelectrode = None
+
+        if platform.system() == "Windows":
+            # hoc.so does not exist on NEURON Windows
+            # although \\hoc.pyd can work here, it gives an error for
+            # nrn_nobanner_ line
+            self.disable_banner = False
+            self.banner_disabled = False
+        else:
+            self.disable_banner = True
+            self.banner_disabled = False
+
+        self.mechanisms_directory = mechanisms_directory
+        self.neuron.h.load_file('stdrun.hoc')
+
+        self.cvode_active = cvode_active
+        self.cvode_minstep_value = cvode_minstep
+
+        self.random123_globalindex = random123_globalindex
+
+    @staticmethod
+    def _nrn_disable_banner():
+        """Disable Neuron banner"""
+
+        nrnpy_path = os.path.join(imp.find_module("neuron")[1])
+        import glob
+
+        hoc_so_list = glob.glob(os.path.join(nrnpy_path, "hoc*.so"))
+
+        if len(hoc_so_list) != 1:
+            warnings.warn(
+                "Unable to find Neuron hoc shared library in %s, "
+                "not disabling banner" % nrnpy_path
+            )
+        else:
+            hoc_so = hoc_so_list[0]
+            nrndll = ctypes.cdll[hoc_so]
+            ctypes.c_int.in_dll(nrndll, "nrn_nobanner_").value = 1
+
+    @property
+    def neuron(self):
+        """Return neuron module"""
+
+        if self.disable_banner and not self.banner_disabled:
+            NrnSimulator._nrn_disable_banner()
+            self.banner_disabled = True
+
+        import neuron  # NOQA
+        if self.mechanisms_directory is not None:
+            neuron.load_mechanisms(
+                self.mechanisms_directory, warn_if_already_loaded=False
+            )
+
+        return neuron
```

Comment:
This code is also from the `NrnSimulator` class.

I think this part of the code can be extracted as a function that only requires the `mechanisms_directory` as a parameter.
That way;
* it can be used in both simulators
* neuron library level function definitions stays out of the object scope
* inheritance will not be necessary between the simulations

```python
        import neuron  # NOQA
        if self.mechanisms_directory is not None:
            neuron.load_mechanisms(
                self.mechanisms_directory, warn_if_already_loaded=False
            )

        return neuron
```